.github/workflows/release.yml

```diff
@@ -0,0 +1,69 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]*'
+
+jobs:
+
+  tests:
+    uses: ./.github/workflows/test.yml
+
+  build:
+    needs: [tests]
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        if: matrix.os == 'ubuntu-latest'
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Install cibuildwheel
+        run: python -m pip install cibuildwheel==2.15.0
+      - name: Build wheels
+        run: python -m cibuildwheel --output-dir dist
+        env:
+          HNSWLIB_NO_NATIVE: true
+          CIBW_PROJECT_REQUIRES_PYTHON: ">=3.7"
+          CIBW_SKIP: "cp312-* pp* *musllinux*"
+          CIBW_ARCHS_MACOS: "x86_64 arm64"
+          CIBW_ARCHS_WINDOWS: "AMD64"
+          CIBW_ARCHS_LINUX: "x86_64 aarch64"
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: python-package-distributions
+          path: dist
+
+  upload:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v4
+      with:
+        python-version: "3.10"
+    - name: Build sdist
+      run: |
+        python -m pip install .
+        make dist
+    - name: Download wheels
+      uses: actions/download-artifact@v3
+      with:
+        name: python-package-distributions
+        path: dist/
+    - name: Publish to Test PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+        repository-url: https://test.pypi.org/legacy/
+    # - name: Publish to PyPI
```

Comment:
thanks

Reply:
Added PYPI_API_TOKEN, we can uncomment now.

Reply:
done